Libs/MRML/Logic/vtkMRMLApplicationLogic.cxx

```diff
@@ -312,6 +312,93 @@ void vtkMRMLApplicationLogic::PropagateVolumeSelection(int fit)
     }
 }
 
+//----------------------------------------------------------------------------
+void vtkMRMLApplicationLogic::PropagateBackgroundVolumeSelection(int fit)
```

Comment:
There's a lot of duplicated code in these methods - can you factor it out?  Maybe make a new method that takes a bitfield with a boolean for propagating each of the layers.  Then the original method can call it with 0x7 and each of these methods can call it with 0x4, 0x2, 0x1 (with an enum).